appcues/src/main/java/com/appcues/trait/appcues/TargetElementTrait.kt

```diff
@@ -24,7 +22,7 @@ internal class TargetElementTrait(
         const val TYPE = "@appcues/target-element"
     }
 
-    private val selector: ElementSelector? = config.getConfigObject("selector")
+    private val selectorProperties = config.getConfig<Map<String, String>>("selector") ?: mapOf()
```

Comment:
was getConfigObject failing silently? so by changing to getConfig (possible null) and checking later during the viewMatchingSelector is whats doing the trick?

Reply:
Yes, `getConfigObject` ended up failing in the Moshi deserialization layer, right here https://github.com/appcues/appcues-android-sdk/blob/main/appcues/src/main/java/com/appcues/data/remote/appcues/adapters/ElementSelectorAdapter.kt#L16

That exception would get thrown during the experience mapping process. The entire response would then effectively be thrown out, as that error would not be handled and propagated all the way out to the root level `CoroutineExceptionHandler`. To the end user it was just a silent failure if the `selector` on the trait had any unexpected structure, which was hard to diagnose.